Mathlib/CategoryTheory/Limits/Constructions/Over/Products.lean

```diff
@@ -20,11 +20,97 @@ open CategoryTheory CategoryTheory.Limits
 
 variable {J : Type w}
 variable {C : Type u} [Category.{v} C]
-variable {X : C}
+variable {X Y Z : C}
 
-namespace CategoryTheory.Over
+/-!
+## Binary products
```

Comment:
Let's add to the module doc that the binary case is done too, and further that it's done separately to get better defeqs.